Feature/problems api full dashboard

timclifford opened the pull request.

bomoko commented on services/api/src/mocks.js:
```diff
@@ -409,116 +460,43 @@ mocks.Task = (parent, args = {}, context, info) => {
     const recentYear = faker.random.arrayElement(['2019', '2020']);
     const vuln_id = `CVE-${recentYear}-${faker.random.number({min: 1000, max: 99999})}`;
-    const source = faker.random.arrayElement(['Clair', 'Drutiny']);
+    const source = faker.random.arrayElement(['Clair', 'Harbor', 'Drutiny']);
```
Not too much of an issue, but just FYI, at the moment we'll not ever have a 'Clair' source - anything from Harbor is marked as coming from Harbor only.

bomoko commented on services/api/src/mocks.js:
```diff
@@ -547,8 +525,9 @@ mocks.Query = () => ({
   deploymentByRemoteId: () => mocks.Deployment(),
   taskByRemoteId: () => mocks.Task(),
-  allProjects: () => new MockList(9),
+  allProjects: () => new MockList(600),
```
Is it convention in the mocks to just drop in a magic number? (I'm not sure how strict they are?)
This feels like something that might be best in a named const?

timclifford replied:
As there quite a few other mocks that use random values here I thought it makes sense to do that same. Maybe it would be best to have a few different values here to mock displays of different sizes though.

bomoko commented on services/api/src/resources/problem/resolvers.ts:
```diff
@@ -76,6 +141,52 @@ export const addProblem = async (
   return R.prop(0, rows);
 };
 
+/**
```
Need to remove this function I think.

bomoko replied:
(and any references to it)

It's not used, and I removed it from the earlier work.

bomoko commented on services/api/src/typeDefs.js:
```diff
@@ -132,6 +132,13 @@ const typeDefs = gql`
     deleted: String
   }
 
+  type ProblemIdentifier {
```
I'm a little confused by this - I think it might be the name - but is this supposed to uniquely identify a problem _across multiple projects_? That is, provide summary information?

If so, I think this needs to be reworked just a little - let's chat when you have a sec.

timclifford replied:
Yep I think we need to chat about this - essentially this was created because returning all problems (e.g. [Problem]) isn't enough to see which projects/environments have been affected. So rather than adjust the Problem type to include project related fields - this ProblemIdentifier type was created. Since then, it might make sense to add a project field in to the Problem type. But lets chat!

bomoko commented on services/keycloak/start.sh:
```diff
@@ -1489,10 +1489,21 @@ function configure_problems_system {
   echo Creating resource problem
 
-  echo '{"name":"problem","displayName":"problem","scopes":[{"name":"view"},{"name":"add"},{"name":"delete"}],"attributes":{},"uris":[],"ownerManagedAccess":""}' | /opt/jboss/keycloak/bin/kcadm.sh create clients/$CLIENT_ID/authz/resource-server/resource --config $CONFIG_PATH -r ${KEYCLOAK_REALM:-master} -f -
+  echo '{"name":"problem","displayName":"problem","scopes":[{"name":"viewAll"},{"name":"view"},{"name":"add"},{"name":"delete"}],"attributes":{},"uris":[],"ownerManagedAccess":""}' | /opt/jboss/keycloak/bin/kcadm.sh create clients/$CLIENT_ID/authz/resource-server/resource --config $CONFIG_PATH -r ${KEYCLOAK_REALM:-master} -f -
```
This is going to need to move into its own function - by the time your code runs, the problems system will have already have run, and therefore this probably won't run again

timclifford replied:
Ah yes - although this is something I wanted to chat about. Is this global check even needed? There are checks on to view each problem and project/environment already.

bomoko commented on services/ui/package.json:
```diff
@@ -39,10 +39,11 @@
     "react-modal": "^3.8.1",
     "react-nice-dates": "^1.0.2",
-    "react-select": "^2.1.1",
+    "react-select": "^3.0.0",
```
Does this change the effective installed version at all?
I don't think we have to worry - I just want to know if there are ramifications to pushing up the major version?

timclifford replied:
![image](https://user-images.githubusercontent.com/2339381/84885883-8ebe4b80-b08b-11ea-9b7c-1f38ba867748.png)
the existing selects work nicely still - this allows us to use some of the latest async stuff as well now

bomoko commented on services/ui/src/components/Accordion/index.js:
```diff
@@ -0,0 +1,105 @@
+import React, { useState, Fragment } from "react";
```
So - I want to make sure of this, are these new components you're adding to the root namespace generic enough to _be_ in the root namespace? The last accordion I moved as a sub-element of the problems space because there were specific assumptions made about its use.

timclifford replied:
After another think about this - I've pull out the component specific stuff for this core accordion and moved all the logic that modifies the columns into the parent components. The only think left is some css that needs to be set for define column widths, but I don't think that is a problem?

bomoko commented on services/ui/src/components/Honeycomb/index.js:
```diff
@@ -0,0 +1,189 @@
+        y: parseInt(display.hexSize * layout.height)
+    };
+
+    const handleHexClick = (project) => {
```
This here seems to suggest that this Honeycomb component is pretty closely tied with the project and environments stuff - which suggests to me we should better compartmentalize (as per my earlier comment)

timclifford replied:
👍 yep - moved this to inside problemsByProject component as it is only going to be used there.

cdchris12 committed:
Small typofix

cdchris12 committed:
Third time's a charm

cdchris12 committed:
This is a commit message

Schnitzel committed:
correct empyDir

Schnitzel committed:
Merge pull request #1968 from shreddedbacon/hotfix-rabbit-dc

Fix indentation for annotations in rabbitmq template

Schnitzel committed:
Merge pull request #1963 from amazeeio/trivy-upgrade

Upgraded Trivy to v0.9.0

Schnitzel committed:
use named targetports

Schnitzel committed:
Merge pull request #1960 from amazeeio/environment-type-labels

Label namespaces with lagoon.sh/environmentType

Schnitzel committed:
Merge pull request #1954 from amazeeio/curl-upgrade

Update curl to `7.70.0-r0` in the PHP images

Schnitzel committed:
Merge pull request #1955 from amazeeio/newrelic-browser-monitoring

Add a new environment variable to control New Relic browser monitoring.

bomoko committed:
Introduces Problems Database System (#1943)

Schnitzel committed:
Merge pull request #1940 from amazeeio/logs-forwarder-fixes

Improve resilience of logs-forwarder

Schnitzel committed:
Merge pull request #1939 from amazeeio/dependabot/npm_and_yarn/apollo-server-express-2.14.2

Bump apollo-server-express from 2.9.4 to 2.14.2

Schnitzel committed:
Merge pull request #1926 from justinlevi/feature/billing-ui-updates

Billing UI Updates

Schnitzel committed:
Merge pull request #1937 from amazeeio/github-community

Moving contributing and CoC to docs root.

Schnitzel committed:
Merge pull request #1953 from amazeeio/harbor-robot-duration

Updated default robot token duration to 500 days.

Schnitzel committed:
Merge pull request #1892 from shreddedbacon/auto-idler-cron

Auto-idler cronjobs

Schnitzel committed:
Merge branch 'master' into version-updater-helper

Schnitzel committed:
Merge pull request #1972 from amazeeio/1971-routes-tls-checker-script

Bash script to check and update TLS broken routes

Schnitzel committed:
Merge pull request #1853 from xantrix/k3s-docs

docs: Kubernetes installation

Schnitzel committed:
disable deleting of harbor projects for now

Schnitzel committed:
Merge branch 'master' into harbor_fix_add_project

Schnitzel committed:
Merge pull request #1961 from amazeeio/varnish_controlport

Adding varnish control port to k8s helmfiles

twardnw committed:
Add support for stakater/ingressmonitorcontroller (#1872)

twardnw committed:
#1970 - add missing task to RC

Schnitzel committed:
Merge pull request #1852 from amazeeio/version-updater-helper

First iteration of Lagoon Version Update Helper

Schnitzel committed:
Merge pull request #1973 from amazeeio/1970-rc-k8s-complete-task

Add missing `builddeploy-kubernetes:complete` task to RC notifications

Schnitzel committed:
use overrides instead of overwrite

Schnitzel committed:
Merge pull request #1776 from amazeeio/harbor_fix_add_project

Update project logic for Harbor projects

Schnitzel committed:
Merge branch 'master' into k8s-active-standby

Schnitzel committed:
Merge pull request #1918 from amazeeio/lagoon-logging-openshift

Make lagoon-logging helm chart OpenShift compatible

Schnitzel committed:
Merge pull request #1772 from shreddedbacon/esclient-typo

Fix typo in esClient

Schnitzel committed:
Merge branch 'master' into varnish-dont-cache-large-binary-files

Schnitzel committed:
Merge branch 'master' into logs-concentrator

Schnitzel committed:
Merge pull request #1768 from amazeeio/shared-to-shared-linux

Improve shared-to-shared-migrate script

Schnitzel committed:
Merge pull request #1959 from amazeeio/logs-concentrator

Logging overhaul

Schnitzel committed:
PERSISTENT_STORAGE_NAME does not exist for these services

use SERVICE_NAME instead

Schnitzel committed:
#1975 typo

Schnitzel committed:
fix issue with emails containing uppercase letters

Schnitzel committed:
Merge pull request #1755 from thom8/composer-patches-docs

Updated docs for lagoonizing a repo with a patches directory.

Schnitzel committed:
fix broken build

Schnitzel committed:
Merge branch 'master' into varnish-dont-cache-large-binary-files

Schnitzel committed:
Merge pull request #1967 from amazeeio/1964-twig-emptydir

set up emptyDir volume for twig render cache

Schnitzel committed:
Merge branch 'master' into develop

Schnitzel committed:
Merge pull request #1920 from shreddedbacon/k8s-active-standby

Support for Active/Standby in Kubernetes

Schnitzel committed:
Introduce MARIADB_DATA_DIR and MARIADB_COPY_DATA_DIR_SOURCE
They allow to change the mariadb data dir or to copy a data dir on start

Schnitzel committed:
Introduce SOLR_DATA_DIR and SOLR_COPY_DATA_DIR_SOURCE
    They allow to change the solr data dir or to copy a data dir on start

Schnitzel committed:
fix wrong naming

Schnitzel committed:
Merge branch 'master' into varnish-dont-cache-large-binary-files

Schnitzel committed:
correct order of symlink

bomoko committed:
Adds healthz-php to php fpm

Schnitzel committed:
Merge pull request #1784 from amazeeio/varnish-dont-cache-large-binary-files

Disable caching of large binary files in Varnish

Schnitzel committed:
fix not existing template

Schnitzel committed:
fix broken if

shreddedbacon committed:
add option to allow pullrequest routes to be autogenerated if autogenerated routes are disabled

shreddedbacon committed:
add branch_regex for autogenerated routes

shreddedbacon committed:
update documentation

shreddedbacon committed:
update documentation

shreddedbacon committed:
set a blank fallback to use what ever is set in enabled

Schnitzel committed:
Merge pull request #1978 from amazeeio/prepopulate-mariadb-solr

Capability to prepopulate mariadb and solr on start

Schnitzel committed:
Merge pull request #1977 from amazeeio/develop

persistentStorageName fixes and other nits

shreddedbacon committed:
fix default regex

shreddedbacon committed:
move the branch definition to environments block instead and check here if the autogenerated route should be created

shreddedbacon committed:
styling of yaml

rocketeerbkw committed:
Add newrelic to api

vincenzodnp committed:
Added some initial checks, support for rocketchat, and shellcheck

Schnitzel committed:
Merge pull request #1984 from amazeeio/1971-routes-tls-checker-script

Added some initial checks, support for rocketchat, and shellcheck

Schnitzel committed:
Merge pull request #1983 from amazeeio/api-newrelic

Add newrelic to api

Schnitzel committed:
v1.7.0 #BlackLivesMatter

Commit:
Problems section per project, problems dashboards and webhooks integration

shreddedbacon committed:
Merge branch 'master' into autogenerate-pr

smlx committed:
Enable annotations on the logs-concentrator service

Schnitzel committed:
#1987 hotfix: correct escaping of slashes

vincenzodnp committed:
Fix notification of error message to rocketchat/slack

Schnitzel committed:
Merge pull request #1988 from amazeeio/hotfix/solr-escaping

#1987 hotfix: correct escaping of slashes

Schnitzel committed:
Brainfuck - v1.7.1

bomoko committed:
Uses kubernetes namespace name to identify environment

bomoko committed:
Merge remote-tracking branch 'upstream/master' into feature/add_healthcheck_endpoint_to_nginx_php

bomoko committed:
Moves healthcheck build into staged build

shreddedbacon committed:
only show the button on the standby environment, and when creating the task, add it to the standby environment tasks list as this is where the UI redirects

shreddedbacon committed:
same logic as route/label showing

Schnitzel committed:
better naming

Schnitzel committed:
Merge branch 'master' into as-task-redirect

Commit:
Problem permission check if user can access project

vincenzodnp committed:
Removed eval

justinlevi committed:
initial redis add

justinlevi committed:
adding in redis nodejs library

justinlevi committed:
hello world

dasrecht committed:
Update Composer to v1.10.8
Update NewRelic to v9.11.0.267

shreddedbacon committed:
add some notes around the switch

shreddedbacon committed:
reword a few bits

justinlevi committed:
Initial Alpha Redis Cache

justinlevi committed:
kinda working with redis

Schnitzel committed:
Merge pull request #1990 from shreddedbacon/as-task-redirect

Minor active/standby adjustments

vincenzodnp committed:
Merge branch 'master' into 1971-routes-tls-checker-script

justinlevi committed:
refactoring to use hashes a bit bettter

bomoko committed:
Adds Luascript healthz fallback

shreddedbacon committed:
feedback tweaks

justinlevi committed:
refactoring out to a redisClient

Commit:
Merge branch 'master' into feature/problems_api_full_dashboard

smlx committed:
Fix typo in variable name

This enables prerollout disabling from environment variables.

cdchris12 committed:
Oops, I a set of quotes

ChandeepKhosa committed:
Update solr.md with some helpful additions

vincenzodnp committed:
Added DEBUG option to print debug information

rocketeerbkw committed:
Fix nslookup returns non-zero error code on 'host.docker.internal'

Commit:
Some small refactoring

Commit:
Merging changes

bomoko committed:
Adds fallback logic for Lua into entrypoint

bomoko committed:
Changes base healthcheck directory

bomoko committed:
Changes mv to cp for locations, changes base rewrite for lagoonhealthz

bomoko committed:
Changes php rewrite rule for healthcheck

bomoko committed:
Fixes regex on lagoonhealthzphp

bomoko committed:
Merge branch 'master' into feature/add_healthcheck_endpoint_to_nginx_php

smlx committed:
Always exclude the lagoon-logging namespace from clusterflow

This avoids log routing loops when debugging log flow via stdout.

smlx committed:
Bump the logging-operator chart dependency

smlx committed:
Add openshift project network global setting to lagoon-logging README

justinlevi committed:
deleting user cache

smlx committed:
Make all components of lagoon-logging chart HA

bomoko committed:
Updates healthz-php version to 0.0.3

smlx committed:
Add fsGroup required by openshift

smlx committed:
Allow endpoints field to be empty for logs-tee services

dasrecht committed:
Update to new webhook url

justinlevi committed:
Fixing inconsistent abstraction

justinlevi committed:
cleanup

bomoko committed:
Adds fast health check (#10)

bomoko committed:
Fixes documentation to reflect correct endpoint

bomoko committed:
Changes health check builder to use the composer image, adds environment variable documentation

rocketeerbkw committed:
Allow setting REDIS_PASSWORD environment variable to protect lagoon redis base images

rocketeerbkw committed:
Update docs for redis environment variables

smlx committed:
Bump lagoon-logging chart version

smlx committed:
Update for lagoon-logs-concentrator to use internal CA

vincenzodnp committed:
Added NOTIFYONLY option

Added DEBUG option
Changed how PROJECTNAME variable is get

vincenzodnp committed:
Fixed the downgit link

Schnitzel committed:
Merge pull request #2022 from amazeeio/2021-fix-link-in-documentation

Fixed the downgit link

bomoko committed:
Merge branch 'master' into feature/add_healthcheck_endpoint_to_nginx_php

smlx committed:
Update chart index

smlx committed:
Bump minreplicas for logs-concentrator

vincenzodnp committed:
Minor fixes

twardnw committed:
provide separate query for looking up billingGroup and status page ID

justinlevi committed:
Translation issues

twardnw committed:
do the same for openshiftbuilddeploy too

twardnw committed:
#1976 - always explicitly use the k3s-lagoon kubeconfig

bomoko committed:
Merge branch 'master' into Changes_problems_to_use_openshift_namespace

cdchris12 committed:
Updating kubectl-build-deploy-dind to use the correct tag when deploying image SHA's with more than one tag associated with them.

cdchris12 committed:
Removing testing print statements

Schnitzel committed:
Merge pull request #2028 from amazeeio/segment-groups-lookup

Segment groups lookup

bomoko committed:
Adds openshift project name lookup to Drutiny problems processor

smlx committed:
Exclude more system namespaces by default

smlx committed:
Disable logs-tee services by default

smlx committed:
Improvements to logs-dispatcher fluentd configuration

* separate index name components with underscores
* fix capitalisation of environmentType tag
* add ability to disable hostname verification for upstream
  logs-concentrator
* implement lagoon-logs-* flow

smlx committed:
Bump logs-concentrator chart version

dasrecht committed:
Merge pull request #2016 from amazeeio/new-webhook-url

Update to new webhook url

twardnw committed:
#1976 - also add the context

twardnw committed:
Merge branch 'master' into 1976-makefile-kubectl-k3s-lagoon

shreddedbacon committed:
add api-development make command

rocketeerbkw committed:
Merge remote-tracking branch 'origin/redis-support-requirepass' into 1908-api-slow

rocketeerbkw committed:
Get redis connection info from environment variables

rocketeerbkw committed:
Cache authz allow in redis. Don't error on redis errors. Code cleanup

smlx committed:
Fix lagoon-logging chart on k8s

Regression caused by adding OpenShift compatible configuration.

rocketeerbkw committed:
Make fewer keycloak requests when using loadGroupsByAttribute

Schnitzel committed:
Use lagoon task system retrying instead of internal retrying
this will not block other builds from being tested and make the retrying more aligned to other services

rocketeerbkw committed:
Cache project groups in redis

rocketeerbkw committed:
Disable apollo server tracing

It was enabled to add data to new relic tracing but it doesn't provide
anything extra and the output gets sent for every request for every
user.

Schnitzel committed:
run 6 task monitors at one per pod

Schnitzel committed:
add ToDo

Schnitzel committed:
make prefetch configurable

bomoko committed:
Removes debug logging from harborScanningCompleted

bomoko committed:
Adds Problems DB OpenShiftProjectName lookup functionality to use project pattern

bomoko committed:
Remove extraneous comment from harborScanningCompleted.ts

bomoko committed:
Merge branch 'master' into Changes_problems_to_use_openshift_namespace

Schnitzel committed:
use proper Numbers

seanhamlin committed:
PATCH existing routes with disable_cookies upon deployment.

Schnitzel committed:
Merge pull request #2039 from amazeeio/disable-cookies-patch

Patch existing routes with disable_cookies upon deployment.

rocketeerbkw committed:
Clear redis caches on relevant mutations

Schnitzel committed:
Merge pull request #2034 from amazeeio/hotfix/parallel-task-monitor

run 6 task monitors at once per pod

smlx committed:
Use image builds from the logging-updates branch in lagoon-logging

rocketeerbkw committed:
Requeue instead of waiting for k8s deployments if they're not next in line

vincenzodnp committed:
Fixed the re-creation of .my.cnf symbolic link

rocketeerbkw committed:
Avoid denying authorization if redis fails

rocketeerbkw committed:
Merge remote-tracking branch 'origin/master' into 1908-api-slow

rocketeerbkw committed:
Use a custom build for api-redis to get deploys working

rocketeerbkw committed:
Add api-redis as make build target

rocketeerbkw committed:
Ensure redis is built before api-redis

smlx committed:
Label using the target env-type instead of the build env-type

shreddedbacon committed:
only run idling if production lagoon

rocketeerbkw committed:
Also clear user caches with group<>project relationships change

rocketeerbkw committed:
Disable newrelic logging to disk in api

rocketeerbkw committed:
Merge branch 'master' into k8squeue-retry

rocketeerbkw committed:
If k8s monitoring services fail for some reason, error out the deployments in lagoon

Schnitzel committed:
Merge pull request #2047 from shreddedbacon/auto-idler

Only run auto-idler service in production Lagoon

rocketeerbkw committed:
Fix getEnvironmentByOpenshiftProjectName returns deleted environment

Schnitzel committed:
Merge pull request #2048 from amazeeio/1908-api-slow

Improve API performance

Schnitzel committed:
Merge pull request #2037 from bomoko/Changes_problems_to_use_openshift_namespace

Change Problems Database population code to use OpenShiftProjectName

Schnitzel committed:
Merge pull request #2029 from amazeeio/k8s_docker_image_selection_fix

Fix for incorrect image selection when images share a SHA hash

Schnitzel committed:
Merge pull request #2026 from amazeeio/bugs/ui-translation-issue

Billing UI - German Translation Issues

Schnitzel committed:
Merge pull request #1999 from amazeeio/another_trivy_fix

Added Quotes for Harbor-Core Configmap

Schnitzel committed:
Merge pull request #1997 from amazeeio/prerollout-disable-environment

Fix rollout task disabling via environment env-vars

Schnitzel committed:
Merge pull request #1992 from amazeeio/composer_newrelic_update

Update Composer to v1.10.8 / Update NewRelic to v9.11.0.267

Schnitzel committed:
Merge pull request #1989 from amazeeio/1971-routes-tls-checker-script

Fix notification of error message to rocketchat/slack

Schnitzel committed:
Merge pull request #1981 from shreddedbacon/autogenerate-pr

Allow additional options for configuring auto generated routes

Schnitzel committed:
Merge pull request #2045 from amazeeio/fix-env-labels

Label using the target env-type instead of the build env-type

Schnitzel committed:
Merge pull request #2032 from shreddedbacon/api-development

Add api-development make command

Schnitzel committed:
Merge pull request #2027 from amazeeio/1976-makefile-kubectl-k3s-lagoon

#1976 - always explicitly use the k3s-lagoon kubeconfig

Schnitzel committed:
Merge branch 'master' into 1088-xdebug-wrong-host

Schnitzel committed:
Merge pull request #1994 from shreddedbacon/as-docs

Documentation updates for Active/Standby

Schnitzel committed:
Merge pull request #2001 from 2Toucans-com/master

Update solr.md with some helpful usability additions

Schnitzel committed:
Merge pull request #2043 from amazeeio/2042-mariadb-fails-to-start-second-time

Fixed the re-creation of .my.cnf symbolic link

seanhamlin committed:
Add PECL Yaml 2.1.0 to the base PHP images

bomoko committed:
Changes FAST_HEALTH_CHECK flag from 'on' to 'true'

Schnitzel committed:
Merge branch 'master' into 1088-xdebug-wrong-host

bomoko committed:
Removes the redundant step of overwriting the lua healthcheck default with a duplicate lua file

bomoko committed:
Removed reference to healthcheck lua file in dockerfile

Schnitzel committed:
Merge pull request #2002 from amazeeio/1088-xdebug-wrong-host

Fix nslookup returns non-zero error code on 'host.docker.internal'

Schnitzel committed:
Merge branch 'master' into feature/add_healthcheck_endpoint_to_nginx_php

Schnitzel committed:
Merge pull request #1979 from bomoko/feature/add_healthcheck_endpoint_to_nginx_php

Adds healthz-php to php fpm

Schnitzel committed:
add redis-password

Schnitzel committed:
Merge pull request #2060 from amazeeio/api-redis-password

add redis-password

Schnitzel committed:
v1.8.0

Schnitzel committed:
Merge branch 'master' into k8squeue-retry

justinlevi committed:
1.8.0 - bug fix for BillingCost query, missing sqlClient for Project Model

Schnitzel committed:
Merge pull request #2061 from amazeeio/bugs/1.8.0-billing-costs-missing-sqlclient

HOTFIX: 1.8.0 - bug fix for BillingCost query, missing sqlClient for Project …

Schnitzel committed:
Merge pull request #2033 from amazeeio/k8squeue-retry

Use lagoon task system retrying instead of internal retrying

Schnitzel committed:
v1.8.1

smlx committed:
Fix NOTES for lagoon-logging chart

smlx committed:
Various improvements to logs-tee service

* make socat unidirectional
* add debug environment variable
* increase transfer block size to maximum for IPv4 UDP
* use broadcasts instead of piping to tee

smlx committed:
Bump logs-dispatcher UDP input buffer

Set to max possible size for IPv4 UDP.

smlx committed:
Remove accidentally committed values file

Schnitzel committed:
Merge pull request #2064 from amazeeio/remove-values

Remove accidentally committed values file

Schnitzel committed:
Merge pull request #2063 from amazeeio/logging-updates

Logging updates

Schnitzel committed:
update tags to master

justinlevi committed:
Cluster name on Billing Group Admin pages

smlx committed:
Roll logs-concentrator statefulset on config change

smlx committed:
Roll logs-dispatcher statefulset on config change

smlx committed:
Bump lagoon-logging and lagoon-logs-concentrator chart versions

smlx committed:
Keep raw nginx log message

We parse the nginx log message into the relevant fields, but some users
may want the original log message.

smlx committed:
Bump logging-operator dependency version

smlx committed:
Bump lagoon-logging chart version

smlx committed:
Add label-namespaces.sh script

vincenzodnp committed:
Changed logic how gather projectname

Set notification channels as an array

rocketeerbkw committed:
Don't cache keycloak authz denials in redis

It's possible the denial is due to a network error or some other
temporary issue

justinlevi committed:
Additional tests and formatting for POLYSITES greater than 10

Schnitzel committed:
Merge pull request #2082 from amazeeio/feature/billing-ui-polysite-groups-of-10

Additional tests and formatting for POLYSITES greater than 10

Schnitzel committed:
Merge pull request #2081 from amazeeio/2078-hotfix-disable-cached-deny

HOTFIX: Don't cache keycloak authz denials in redis

Schnitzel committed:
Merge pull request #2075 from amazeeio/keep-raw-nginx-log

Keep raw nginx log message

Schnitzel committed:
Merge pull request #2077 from amazeeio/label-namespaces-script

Add helper script for labelling namespaces

Schnitzel committed:
Merge pull request #2070 from justinlevi/feature/ui-billing-cluster

Add Cluster Name on Billing Group Admin pages

Schnitzel committed:
Merge pull request #2079 from amazeeio/check_acme_routes-script-enhancements

Some enhancements for check_acme_routes.sh script

Schnitzel committed:
Merge pull request #2057 from amazeeio/yaml-pecl

Add PECL Yaml 2.1.0 to the base PHP images

Commit:
Merge remote-tracking branch 'lagoon/master' into feature/problems_api_full_dashboard

Commit:
Merge conflict with yarn lock after ui changes
